### ISSUE TITLE
TS2315: Type 'EventEmitter' is not generic.

### ISSUE THREAD

**Issue by JanUnld** [WilixLead/iohook#49]
Hey there. I am currently trying to use **iohook** with Typescript. I really don't know if there's just something horrible wrong with me and I'm missing something but I can't get the compiler to accept the `index.d.ts` file the repo provides... the compiler always complains about:

```
TS2315: Type 'EventEmitter' is not generic.
```

It works perfectly fine when I'm executing the compiled source since Typescript doesn't really care about the error anyways. But I really wan't to get rid of that error, since it forces any chained command sequence to fail.

At this state I'm just curious what's happening so I decided to open this issue. Thanks in advance.

**Comment by Djiit** [comment]
Hi there ! Thanks for reporting this. I'm not quite familiar with TypeScript concepts so, if anybody can check this and propose a fix, it would be awesome !

**Comment by JanUnld** [comment]
Alright so after looking a bit deeper into `@types/node` I found that the `EventEmitter` class doesn't accept any type parameters.

This issue doesn't affect the actual functionality provided by **iohook**. The only _bad_ effect you have is, that Typescript can't properly resolve what `EventEmitter` would given us. So some `ts` code like this...

```typescript
ioHook.on("event", e => console.log(e));
```

...would throw some errors telling us that `on` doesn't exist on type `IOHook` proving the stuff I mentioned above. Removing the type parameter from the `IOHook` class would do the trick here.

**Working around Typescript:**

Since Typescript only cares about what we are telling the compiler to care about we could work around this issue just fine changing the example above to...

```typescript
(ioHook as any).on("event", e => console.log(e));
```


Whatever... I've already forked the repo and will try to provide a new version of the `index.d.ts` file soon...
